### PR TITLE
MDS-2738/MDS-3056: Dynamic Image Support for Document Generation (Inspector Signatures) and Permit DRAFT Watermark

### PR DESCRIPTION
**Pull request by exzizt**
# Main
* Creates and implements a "dynamic image support" feature for our document generation system. This was used to sign the generated permits and Notice of Work letters with the Issuing Inspector's signature.
* Places a DRAFT watermark on our generated draft permit documents.
* Users now assign an "Issuing Inspector" on a Notice of Work to be used in signing the permit and other documents.
* Adds the email and phone of the Issuing Inspector to Notice of Work letters
* Shows the signature of the Issuing Inspector on the Administrative Tab and on the Party Profile, if applicable

### DRAFT watermark on a draft permit
image

### Issuing Inspector signature on a permit
image

### Issuing Inspector signature on a letter
image

### Signature/missing signature on the admin tab
image
image

### Signature/missing signature on party profile page
image
image

# Other
* Simplifies the names of our NoW and permit template files.
* Tidies up some code/SQL (removes redundant JSON fields from various document template "form spec JSON" fields, for example).
* Fixes an issue where General Conditions were being displayed in the Geotechnical Conditions section in our generated permits.

### CODE REVIEW

File: services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js [hunk 1]
```diff
@@ -32,17 +35,59 @@ const UpdateNOWLeadInspectorForm = (props) => {
...
+            disabled={!props.isEditMode}
+            onSelect={props.setIssuingInspectorPartyGuid}
+          />
+        </Form.Item>
+        {!props.isEditMode && (
+          <>
+            {props.noticeOfWork?.issuing_inspector?.signature ? (
+              <img
+                src={props.noticeOfWork?.issuing_inspector?.signature}
+                alt="Signature"
```
Comment (mmumby) [inline services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js:71]: Whats going on with the question marks here?  I'm not familiar with this syntax
`props.noticeOfWork?.issuing_inspector?.signature`

**Review by mmumby — COMMENTED** [review]
FE looks great, Luke. No concerns here.
Just one question regarding some unknown syntax.

File: services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js (hunk 1 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js:71]: This is a clever null check.. why haven't we been using this everywhere

Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js:71]: also, if you already checked its existence on line 69, can you just pass it to the src?

Comment (exzizt) [reply services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js:71]: > Just googled it - This is a clever null check.. why haven't we been using this everywhere

I know right!! I learned about it the other night... We definitely should start using it. Sounds good!

Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/UpdateNOWInspectorsForm.js:71]: I'm absolutely going to start using this!

File: services/core-api/app/api/document_generation/models/document_template.py [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+import json, os, docx, io, base64
+
 from flask import current_app
-from datetime import datetime
-from app.extensions import db
 from sqlalchemy.schema import FetchedValue
-from sqlalchemy.orm import mapper
 from sqlalchemy.dialects.postgresql import UUID
 from sqlalchemy.ext.hybrid import hybrid_property
+from docx.shared import Inches
```
Comment (nayr974) [inline services/core-api/app/api/document_generation/models/document_template.py:7]: row by row?

File: services/core-api/app/api/now_applications/models/now_application_document_type.py [hunk 3]
```diff
@@ -27,15 +29,42 @@ def get_with_context(cls, document_type_code, context_guid):
         return document_type
 
     def transform_template_data(self, template_data, now_application):
+        def create_image(source, width=None, height=None):
+            return {'source': source, 'width': width, 'height': height}
 
-        # Transform template data for "Working Permit" (PMT) or "Working Permit for Amendment" (PMA) documents
```
Comment (mhuseinov) [inline services/core-api/app/api/now_applications/models/now_application_document_type.py:31]: As for me PMT and PMA looked helpful here 🙂

**Review by nayr974 — APPROVED** [review]

**Review by mhuseinov — APPROVED** [review]
🔥

File: services/core-api/app/api/document_generation/models/document_template.py (hunk 2 above)
Comment (exzizt) [reply services/core-api/app/api/document_generation/models/document_template.py:7]: I don't get the reference :/

**Pull request merged**